controllers/medusa/backup_controller_test.go

```diff
@@ -0,0 +1,280 @@
+			reflect.DeepEqual(updated.Status.CassdcTemplateSpec.Spec.Users, dc.Spec.Users) &&
+			reflect.DeepEqual(updated.Status.CassdcTemplateSpec.Spec.AdditionalSeeds, dc.Spec.AdditionalSeeds)
+	}, timeout, interval)
+
+	t.Log("verify the backup finished")
+	require.Eventually(func() bool {
+		updated := &api.CassandraBackup{}
+		err := testClient.Get(context.Background(), backupKey, updated)
+		if err != nil {
+			return false
+		}
+		return len(updated.Status.Finished) == 3 && len(updated.Status.InProgress) == 0
+	}, timeout, interval)
+
+	t.Log("verify that medusa gRPC clients are invoked")
+	assert.Equal(medusaClientFactory.GetRequestedBackups(), map[string][]string{
+		fmt.Sprintf("%s:%d", getPodIpAddress(0), backupSidecarPort): {backupName},
+		fmt.Sprintf("%s:%d", getPodIpAddress(1), backupSidecarPort): {backupName},
+		fmt.Sprintf("%s:%d", getPodIpAddress(2), backupSidecarPort): {backupName},
+	})
+}
+
+func createDatacenterPods(t *testing.T, ctx context.Context, dc *cassdcapi.CassandraDatacenter, testClient client.Client) {
+	for i := int32(0); i < dc.Spec.Size; i++ {
+		pod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: dc.Namespace,
+				Name:      fmt.Sprintf("%s-%d", dc.Spec.ClusterName, i),
+				Labels: map[string]string{
+					cassdcapi.ClusterLabel:    dc.Spec.ClusterName,
+					cassdcapi.DatacenterLabel: dc.Name,
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "cassandra",
+						Image: "cassandra",
+					},
+					{
+						Name:  backupSidecarName,
+						Image: backupSidecarName,
+					},
+				},
+			},
+		}
+		err := testClient.Create(ctx, pod)
+		assert.NoError(t, err, "failed to create datacenter pod")
+
+		patch := client.MergeFrom(pod.DeepCopy())
+		pod.Status.PodIP = getPodIpAddress(int(i))
+
+		err = testClient.Status().Patch(ctx, pod, patch)
+		assert.NoError(t, err, "failed to patch datacenter pod status")
+	}
+}
+
+// Creates a fake ip address with the pod's orginal index from the StatefulSet
+func getPodIpAddress(index int) string {
+	return "192.168.1." + strconv.Itoa(50+index)
+}
+
+type fakeMedusaClientFactory struct {
+	clientsMutex sync.Mutex
+	clients      map[string]*fakeMedusaClient
+}
+
+func NewMedusaClientFactory() *fakeMedusaClientFactory {
+	return &fakeMedusaClientFactory{clients: make(map[string]*fakeMedusaClient, 0)}
+}
+
+func (f *fakeMedusaClientFactory) NewClient(address string) (medusa.Client, error) {
+	medusaClient := newFakeMedusaClient()
+	f.clientsMutex.Lock()
+	f.clients[address] = medusaClient
+	f.clientsMutex.Unlock()
+	return medusaClient, nil
+}
+
+func (f *fakeMedusaClientFactory) GetRequestedBackups() map[string][]string {
+	requestedBackups := make(map[string][]string)
+	for k, v := range f.clients {
+		requestedBackups[k] = v.RequestedBackups
+	}
+	return requestedBackups
+}
+
+type fakeMedusaClient struct {
```

Comment:
FYI you can also generate a mock with mockery + testify, we already have a few ones in the `mocks` package.

Reply:
gotcha, didn't do it due to lack of time.